creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -377,37 +436,139 @@ class SnapbackSM {
+   * existing user data for the current user. If there is pre-existing data in the randomly selected node, keep
+   * searching for a node that has no state.
+   *
+   * If an insufficient amount of new replica set nodes are chosen, this method will throw an error.
+   * @param {Object} param
+   * @param {Set<string>} param.unhealthyReplicasSet a set of the unhealthy replica set endpoints
+   * @param {string[]} param.healthyNodes an array of all the healthy nodes available on the network
+   * @param {string} param.wallet the wallet of the current user
+   * @returns a string[] of the new replica set nodes
+   */
+  async selectRandomReplicaSetNodes ({ unhealthyReplicasSet, healthyNodes, wallet }) {
+    let newReplicaNodesSet = new Set()
+    let selectNewReplicaSetAttemptCounter = 0
+    while (newReplicaNodesSet.size < unhealthyReplicasSet.size && selectNewReplicaSetAttemptCounter++ < MAX_SELECT_NEW_REPLICA_SET_ATTEMPTS) {
+      let randomHealthyNode = _.sample(healthyNodes)
 
-      // Select a random node that is not from the current replica set
-      newSecondary = healthyPotentialPeers[Math.floor(Math.random() * healthyPotentialPeers.length)]
+      // If node is already present in set, keep finding a unique healthy node
+      if (newReplicaNodesSet.has(randomHealthyNode)) continue
 
       // Check to make sure that the newly selected secondary does not have existing user state
       try {
-        const clockValue = await CreatorNode.getClockValue(newSecondary, wallet)
-        if (clockValue === -1) currentNodeHasExistingUserState = false
+        const clockValue = await CreatorNode.getClockValue(randomHealthyNode, wallet)
+        if (clockValue === -1) { newReplicaNodesSet.add(randomHealthyNode) }
       } catch (e) {
         // Something went wrong in checking clock value. Reselect another secondary.
-        this.logError(e.message)
+        this.logError(`[selectRandomReplicaSetNode] ${e.message}`)
       }
     }
 
-    if (currentNodeHasExistingUserState) throw new Error(`Unable to select new secondary given current healthy Content Nodes=[${healthyNodes}]`)
+    if (newReplicaNodesSet.size < unhealthyReplicasSet.size) {
+      throw new Error('[selectRandomReplicaSetNode] Not enough healthy nodes found to issue new replica set')
+    }
 
-    const currentHealthySecondary = unhealthyReplicasSet.has(secondary1) ? secondary2 : secondary1
+    return Array.from(newReplicaNodesSet)
+  }
 
-    // NOTE: For v0 implementation, `newSecondary2` will be the newly selected secondary, whilst the other nodes will be the
-    // original replica set primary and secondary1
-    return {
-      newPrimary: primary,
-      newSecondary1: currentHealthySecondary,
-      newSecondary2: newSecondary
+  // If the primary is part of the replica set, truly check to see if it is unhealthy.
+  async finalizeUnhealthyReplicaSet ({ unhealthyReplicasSet, primary, secondary1, secondary2, wallet, secondariesToClockMap }) {
```

Comment:
isn't this logic exactly what happens in `processStateMachineOperation` when it calls `peerSetManager.getUnhealthyPeers()`? that also queryVerboseHealthCheck -  i guess its fine to just re-check it here

Reply:
yes and no:
yes -- `getUnhealthyPeers()` checks the verbose health check route to determine peer health and is wrapped in a try/catch
no -- `getUnhealthyPeers()` has additional peer health checking conditions defined in `determinePeerHealth()` that are much more rigorous than what we want. also, this function's intention is to check peer health, rather than an individual primary

Reply:
resolving bc outdated